polar_training2tcx

```diff
@@ -136,7 +136,7 @@ def output_tcx(parsed)
 
                     xml.Trackpoint {
                       xml.Time (start + elapsed_with_pauses).strftime("%Y-%m-%dT%H:%M:%S.%3NZ")
-                      if route_i && route_samples.duration[route_i] #&& (route_samples.duration[route_i] / 1000 <= elapsed)
+                      if route_i && route_samples.duration[route_i] && (route_samples.duration[route_i].to_f / 1000 <= elapsed)
```

Comment:
If I compare with a Polar Flow TCX export, it looks like that they actually reuse route_samples.latitude|longitude as long as `route_samples.duration[route_i].to_f / 1000 <= elapsed`.
I just pushed a different fix that gives me the exact same trackpoints as Polar Flow on my latest ride. Please give it a try and check that it works for you too!

Reply:
I tried it with the same tracks I used before - but no, it does not work as I would need, and neither it matches the PolarFlow TCX files. 

Sometimes I start running without waiting for GPS to be found, and so quite a lot of initial trackpoints may have data for distance, cadence etc., but do not have position data. With my patch, it works well, the positions appear late in the track. With your fix, all the positions are still "shifted" to the beginning (which is strange, as the code suggests that instead the first position should be reused again and again) and disappear at the end of the track.

Reply:
@cmaion, I have figured out why it did not work as i expected; looks like the condition for `route_i` increment contained an unfortunate typo. I have updated this patch, could you please re-check?